### DISCUSSION TITLE
Mouse events doesn't work with FACEIT Anti-Cheat enabled (CS2)

### DISCUSSION THREAD

**Discussion by dvgmdvgm** [wasiejen/Free-Snap-Tap#38] [General]
Hello! Everything works greate but if i want to play CS2 on FACEIT platform i have to run FACEIT AC and if AC is enabled my `mouse_left, -mouse_right` completely broken. Any idea how to fix this? Maybe different input method?

**Comment by wasiejen** [comment]
Heyho, welcome back :-D

What do you mean by "broken"? Just not working? Or doing something else?

The only thing that we could try is to include another package to send these mouse events instead of the actual used package pynput and see if any of them will work. But since I have no FACEIT AC on my PC, I can not test it myself, which makes it hit and miss.

**Comment by s0cr4tes** [comment]
Can you download the esportal anti cheat e.x and check there aswell? It works perfectly outside the anti cheat, but when i use AC on Esportal or Faceit, it is countered some[[esportal-client-installer_en.zip](https://github.com/user-attachments/files/16727526/esportal-client-installer_en.zip)_en.zip…]()how. 

It works perfectly outside the anti cheat, but when i use AC on Esportal or Faceit, it is countered somehow.

**Comment by dvgmdvgm** [comment]
> Heyho, welcome back :-D
> 
> What do you mean by "broken"? Just not working? Or doing something else?
> 
> The only thing that we could try is to include another package to send these mouse events instead of the actual used package pynput and see if any of them will work. But since I have no FACEIT AC on my PC, I can not test it myself, which makes it hit and miss.

Any keyboard aliases working great with AntiCheat enabled but `mouse_left, mouse_right` completely ignored.
I will upload some video

**Comment by dvgmdvgm** [comment]
@wasiejen https://www.youtube.com/watch?v=S85mbh_1Vwg

**Comment by wasiejen** [comment]
Thanks for the video. Nice to see it in action.
I am impressed how fast you could switch into the game and create a new one :-D

The new version 0.8.3 could help because of the threading of Aliases ... but don't get your hopes up. 
I did not implement this for that issue, but to prevent interference with real input and better suppression of key used to trigger the aliases.

**Comment by wasiejen** [comment]
And for easier reloading on v0.8.2 and later and if you are using the controls implemented:
when you pause and resume the script via DEL - on resume the groups will be reloaded from the files.

**Comment by dvgmdvgm** [comment]
> Thanks for the video. Nice to see it in action. I am impressed how fast you could switch into the game and create a new one :-D
> 
> The new version 0.8.3 could help because of the threading of Aliases ... but don't get your hopes up. I did not implement this for that issue, but to prevent interference with real input and better suppression of key used to trigger the aliases.

mouse events still ignored in 0.8.3. keboard events worling greate

**Comment by wasiejen** [comment]
Ok, that would have been to easy :-D

Will have a further look into it.
Will start a new branch and test out some other python input libraries in it.
But it will be slow going - I will not have much time in the next days. ^^

**Comment by wasiejen** [comment]
Can you test out https://github.com/wasiejen/Free-Snap-Tap/blob/alternative_mouse_controller/free_snap_tap.py?
Or see https://github.com/wasiejen/Free-Snap-Tap/releases/tag/v0.8.3-testbuild for the exe.
For the left mouse button I implemented a DirectInput Lib. 
**Only the left mouse button** yet xD.
But that is enough to test if it even works :-D Pls give feedback if it works or not. thanks ^^

So, times flies. 
Have to go now :-D

**Comment by dvgmdvgm** [comment]
> Can you test out https://github.com/wasiejen/Free-Snap-Tap/blob/alternative_mouse_controller/free_snap_tap.py? Or see https://github.com/wasiejen/Free-Snap-Tap/releases/tag/v0.8.3-testbuild for the exe. For the left mouse button I implemented a DirectInput Lib. **Only the left mouse button** yet xD. But that is enough to test if it even works :-D Pls give feedback if it works or not. thanks ^^
> 
> So, times flies. Have to go now :-D

Unfortunately not working(

**Comment by wasiejen** [comment]
Ok next mouse controller.
Uses directly the windll over ctypes.
left, right and middle mouse now uses this.

Same file location as before:
https://github.com/wasiejen/Free-Snap-Tap/blob/alternative_mouse_controller/free_snap_tap.py

**Comment by dvgmdvgm** [comment]
> windll

Nah. It's not working. 
It' sokay i can ude keybinding) keep focus on the next features Man

**Comment by wasiejen** [comment]
@dvgmdvgm: You might like the next small feature update with V0.8.4 ;-D

**Comment by dvgmdvgm** [comment]
> @dvgmdvgm: You might like the next small feature update with V0.8.4 ;-D

Yeah! Window focus is greate feature!
What do you think about aliases?
I mean if i didn't set delay inside of txt file it can be presed at the same time? And if daelay was setted by txt file it will pressed 1by1 with delay? WDYT?
For example:
`n, space, w, o`  if n was pressed all three buttons should be pressed at the same time.
`n, space, w, o|30` space & w pressed at the same time and o pressed after 30ms of space&w was pressed.

**Comment by wasiejen** [comment]
> Yeah! Window focus is greate feature! What do you think about aliases? I mean if i didn't set delay inside of txt file it can be presed at the same time? And if daelay was setted by txt file it will pressed 1by1 with delay? WDYT? For example: `n, space, w, o` if n was pressed all three buttons should be pressed at the same time. `n, space, w, o|30` space & w pressed at the same time and o pressed after 30ms of space&w was pressed.

moin moin,

you can already do that by adding |0|0 to every key you want to be send without any delay. ^^ 
A Random Delay between 0 and 0 is still 0 :-D
`n, space|0|0, w|0|0m o|0|0` would play the sequence without any delay.

maybe i could activate the seperate default alias_min/max_delays and make them accessible over start-arguments. In the py-file you can already edit ALIAS_MIN_DELAY_IN_MS and ALIAS_MAX_DELAY_IN_MS and then the key_groups will have different default delays than the tap groups. e.g. 0,0 would then mean without delay and you have to define delays via | notation.

:-)
